substrate/frame/examples/basic/src/lib.rs

```diff
@@ -38,18 +38,32 @@
 //!
 //! This pallet provides basic examples of using:
 //!
+//! - Pallet hooks to implement some logic that's executed before and after a block is imported
```

Comment:
> before and after a block is imported

Is wrong. these hooks are executed at the start and end of the block oimport, so they are part of it.

Everything in https://paritytech.github.io/polkadot-sdk/master/frame_support/traits/trait.Hooks.html#summary is part of block import.